pytorch_lightning/trainer/progress.py

```diff
@@ -0,0 +1,82 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Progress:
+    """ Basic dataclass to track event progress.
+
+    Args:
+        ready: Intended to track the number of batches fetched from the dataloader.
+        started: Intended to be incremented after `on_*_batch_start` completes.
+        processed: Intended to be incremented after `*_step` runs.
+        completed: Intended to be incremented after `on_*_batch_end` runs.
+    """
+    ready: int = 0
+    started: int = 0
+    processed: int = 0
+    completed: int = 0
+
```

Comment:
@carmocca for now i haven't included the optimization progress. i can add it in a follow up PR. lmk what you think of these changes

Reply:
Thanks Ananth, it's definitely better but this is what I had in mind. Please tell me what you think and feel free to cherry-pick the commit into this PR.

https://github.com/PyTorchLightning/pytorch-lightning/commit/79d2990d0a6d8d927287dff6c9b124d9f6c71441

Haven't fixed the tests. Some names could be changed and obviously it's missing utility functions but I feel like that is good enough

Reply:
Created this document to explain what each of these would mean, this way you can comment

https://docs.google.com/document/d/1U73b4uFML2MNdObh6MVsBpFDrgLUOU2Sd_x0_DVk-Bo/edit?usp=sharing

cc @awaelchli :)